core/Cargo.toml

```diff
@@ -39,8 +40,9 @@ serde_json = "1.0"
 sha-1 = "0.9"
 shannon = "0.2.0"
 thiserror = "1.0.7"
-tokio = { version = "1.0", features = ["io-util", "net", "rt", "sync"] }
+tokio = { version = "1.5", features = ["io-util", "macros", "net", "rt", "time", "sync"] }
 tokio-stream = "0.1.1"
+tokio-tungstenite = { version = "0.14", default-features = false, features = ["native-tls"] }
```

Comment:
I'm afraid there's no way to add websocket support without new dependencies.

`native-tls` failed in CI. I hesitated to use `rustls` because it brings up a new question: `rustls` uses the `ring` crate for cryptography. We could use the `ring` crate too for our crypto stuff to avoid duplication, but we won't be able to switch completely, e.g. because it doesn't support aes192.

Reply:
Unfortunately not: https://github.com/briansmith/ring/issues/112

Reply:
That's a pity. The statement "Because nobody is using it in the real world." is debatable :wink:

Reply:
I was thinking, "what does `hyper-tls` use"? And I see it's `nativetls`...

Reply:
Here you go: https://crates.io/crates/hyper-rustls

Reply:
`rustls` wouldn't be a bad choice: https://www.abetterinternet.org/post/preparing-rustls-for-wider-adoption/

I don't really care.

Reply:
All valid arguments. Guess you can argue either way. Not partial to either here too, just thinking along how to make this as lean and maintainable as possible.

Reply:
I learned that `tonic`(grpc framework based on hyper) supports only rustls. 

Thankfully, it's very easy to switch.